src/xilefunds.js

```diff
@@ -0,0 +1,131 @@
+            }
+            msg = msg + "```"
+            const StocksEmbed = new Discord.MessageEmbed()
+                .setColor("#0368f8")
+                .setTitle("Xilefunds' stock market")
+                .setDescription(msg)
+                .setTimestamp()
+            if (Stocks.auction.seller) {
+                StocksEmbed.addField("Current auction:", `Seller: ${Stocks.auction.seller}\nLatest offer: ${Stocks.auction.buyer ? (Stocks.auction.buyer + " -> " + Stocks.auction.price) : "`none`"}`)            
+                StocksEmbed.addFooter("The auction will end in " + Time.converTime(Math.floor(Stocks.timeout._idleTimeout - ((process.uptime() * 1000) - Stocks.timeout._idleStart))))
+            } else StocksEmbed.addFooter("Start an auction to sell your Xilefund!")
+            message.channel.send(StocksEmbed)
+            return
+        }
+        case "offer": {
+            if (Stocks.auction.seller == undefined) {
+                message.channel.send("There isn't any auction at the moment, start your own!")
+                return
+            } else if (Stocks.auction.seller == message.author.username) {
+                message.channel.send("You can't offer in your own auction!")
+                return
+            }
+            const offer = parseInt(args[1])
+            if (isNaN(offer)) {
+                message.channel.send("That is not a valid offer")
+                return
+            } else if (offer <= Stocks.auction.price) {
+                message.channel.send("Your offer is too low, you have to offer something higher than the current offer (" + Stocks.auction.price + ")")
+                return
+            } else if (EconomySystem.money < offer) {
+                message.channel.send("You can't offer money that you don't even have!")
+                return
+            }
+            Stocks.auction.buyer = message.author.username
+            Stocks.auction.buyerid = message.author.id
+            Stocks.auction.price = offer
+            message.channel.send("You successfully offered " + offer + " to this auction!")
+            return
+        }
+        case "auction": {
+            if (Stocks.auction.seller != undefined) {
+                message.channel.send("There is already another auction going at the moment")
+                return
+            } else if (EconomySystem.xilefunds < 1) {
+                message.channel.send("You don't have any Xilefund to auction!")
+                return
+            }
+            const price = args[1] || Stocks.value
+            if (price < 1) {
+                message.channel.send("uhh...you can't give this away for free...")
+                return
+            }
+            Stocks.auction.seller = message.author.username
+            Stocks.auction.sellerid = message.author.id
+            Stocks.auction.price = price
+            Stocks.timeout = setTimeout(() => {
+                const SellerDMChannel = message.author.createDM()
```

Comment:
`User#createDM` returns a promise. You need to `.then` or `await` the value first.

Reply:
oh alright